### ISSUE TITLE
false negative if template is long

### ISSUE THREAD

**Issue by billzt** [gjeunen/reference_database_creator#15]
Step 4.1
```
crabs insilico_pcr --input input.fasta --output output.fasta --fwd AGTC --rev ACTG --error 4.5
```

Suppose the length of expected amplicon is 100bp, and there is a long template (e.g. a ~16Kb MT) in `input.fasta`, where its sequence is like:
```
.....some_seq....[fwdPrimer]...regionA:999bp....[revPrimer]....some_other_seq......[fwdPrimer]...regionB:100bp...[revPrimer].....some_other_seq...
```

regionA is false amplicon and regionB is true amplicon. However cutadapter is designed to cut "reads" and it can not deal with such situation correctly. Usually it would output the fake 999bp-regionA to `output.fasta`.

Then this fake amplicon is used as "seeds" to do PGA in step 4.2, further collecting other long amplicons. And then in step 7.1 these long amplicons were all filtered out due to exceeding `--maxlen`. And the result is that this template has no amplicons.

In fact, it has true amplicon B.

**Comment by gjeunen** [comment]
Hello @billzt,

Thank you very much for your comment. A simple solution to your query would be to run the filter parameters after the in silico PCR. This would remove the longer "fake" amplicon. Afterwards, you can run the PGA function. As the "fake" amplicon sequence isn't in the seed sequences anymore, the "true" amplicon will be found by the PGA. 

Currently, you can use the `tax_format` and `db_import` functions to change the documents to the correct format for the input file. This is a bit of a workaround though and I'll change some of the code to make it easier to switch from one function to the other. I'm thinking of ditching the ".fasta" format for the first couple of functions and go straight to a tab-delimited structure for all files in CRABS. That way you can freely use any of the CRABS functions in any order. I'm currently on leave though, so these changes will only come into effect when I'm back at work.

Best,
Gert-Jan

**Comment by billzt** [comment]
Yes, perhaps this is a good solution

**Comment by gjeunen** [comment]
Hello @billzt,

Yes, in this case, you would need to use `--filter_method relaxed`. Thank you for the idea to add in a figure. I will add this to my to do list as well.

Best,
Gert-Jan

**Comment by billzt** [comment]
OK, I understand. Therefore `--filter_method relaxed` is important in this situation, although in the literature it said `--filter_method  strict` is recommended.

**Comment by billzt** [comment]
Therefore, I think a reasonable practice would be

1. Narrow your  `input.fasta` region. For example, if your primer locates in 12S rRNA, then only use 12S rRNA sequences as  `input.fasta`. Avoid using whole genome
2. Run Step 4.1 `crabs insilico_pcr`
3. Filter the output based on length and N-containg reads using an own script
4. Run Step 4.2 `crabs pga`, using parameters `--filter_method relaxed` and `--speed slow`. Setting `--percid` to a low identity such as 0.7
5. Filter the output based on length and N-containg reads again
6. Add taxID for each sequence (I used an own script)
7. check the last 10bp in 3'end of primer-template binding regions. Delete those with many mismatches. I found there were ~0.5% sequences with >1 mismatches in the last 10bp in 3'end of primer-template binding regions.
8. Dereplicate
